scrape_all_prs.py

```diff
@@ -29,24 +32,24 @@ def rotate_token():
 # Function to handle rate limits by rotating tokens or waiting if all are exhausted
 def handle_rate_limit(g):
     rate_limit = g.get_rate_limit().core
+
     if rate_limit.remaining == 0:
         reset_time = rate_limit.reset
         wait_time = (reset_time - datetime.now(timezone.utc)).total_seconds()
-        print(f"Token {current_token_index + 1} rate limit exceeded. Waiting {wait_time:.2f} seconds.")
+        print(f"Token {current_token_index + 1} rate limit exceeded.")
         
         # Rotate to the next token or wait if all tokens are exhausted
         if current_token_index == len(tokens) - 1:
             print(f"All tokens exhausted. Waiting {wait_time:.2f} seconds for reset.")
             time.sleep(wait_time)
-            return get_github_instance()
```

Comment:
This was waiting, but then returning the same last token, so would always only be taking advantage of last token. Hence always returning `rotate_token()` (if waited or not)